### DISCUSSION TITLE
Error while installing prisma

### DISCUSSION THREAD

**Discussion by DeveloperParth** [pnpm/pnpm#7076] [General]
This happened after I ran store prune command because there was an error related to @aws-sdk/s3-request-presigner the error was at build times the error was present in local machine as well as on Vercel
```shell
Module not found: Can't resolve '@aws-sdk/signature-v4-crt' in '/vercel/path0/node_modules/.pnpm/@aws-sdk+signature-v4-multi-region@3.398.0/node_modules/@aws-sdk/signature-v4-multi-region/dist-cjs'
``` 
so I thought maybe clearing store might help but clearing store worsened things as now when i tried to run ```pnpm i``` it would stop installing packages due to same error
```shell
 ERR_PNPM_LINKING_FAILED  Error: EPERM: operation not permitted, rename '\node_modules\.pnpm\@prisma+client@5.0.0_prisma@5.0.0\node_modules\@prisma\client_tmp_13528' -> '\node_modules\.pnpm\@prisma+client@5.0.0_prisma@5.0.0\node_modules\@prisma\client'
```
Its issue with only prisma package as running ```pnpm remove prisma``` installs everything correctly.

While main issue is indeed the second one with prisma I am not sure the first issue with aws-sdk is solved so itd be great if someone can help with that too.

**Comment by Jolg42** [comment]
- By `pnpm remove prisma` did you mean `pnpm remove @prisma/client` actually? That would make more sense to me here 🤔 
- What's your current pnpm version? 
- What's your local operating system?
- Are you running pnpm inside docker?
- Have you tried to delete the cache already? Maybe try a manual removal with `rm -r ~/Library/pnpm/store`
- Have you tried to delete and re-create the `pnpm-lock.yaml` file?

**Reply by DeveloperParth** [reply]
- I should have clarified it. So in an existing project running ```pnpm remove prisma``` works and installs everything, after your response I tried running ```pnpm remove @prisma/client``` while prisma still present in package.json and this doesn't work and throws the same error.
- Ive tried it with different versions ```8.7.5``` ```8.7.4``` ```8.7.0```
- Windows 11
- No
- Yes I've tried to delete cache multiple times. Ive also reinstalling pnpm and also reinstalled and updated nodejs from ```18.17.0``` to ```18.17.1```
- yes Ive tried deleting the file.

Somethings ive tried after making this thread are

- Making a completely fresh project with ```pnpm init``` and then adding prisma but it doesnt 
- Deleting node_modules from other projects and installing them but same error. Prisma wont install on my system with pnpm

Edit:Also after the corrupt instillation there would be only .pnpm folder inside node_modules and nothing else

**Reply by Jolg42** [reply]
Interesting 🤔 So you are running pnpm on Windows 11 using WSL 2 I guess?

**Reply by DeveloperParth** [reply]
No Im running it directly

**Comment by Jolg42** [comment]
@DeveloperParth Concerning the Prisma issue @janpio is curious to get some reproductions steps and try out.
Is it as simple as this?
- create empty directory
- pnpm init
- pnpm add prisma

Or is there anything else needed to trigger this?

**Reply by janpio** [reply]
(I am using Windows... so I have been "volunteered" to try this out if we have a reproduction)

**Reply by DeveloperParth** [reply]
Well yes that was what happening but I while trying to figure it out i tried running ```pnpm add prisma@latest``` and well it worked.
Now for some project it installs and for some it doesn't.

I tried this example [https://github.com/prisma/prisma-examples/tree/latest/typescript/rest-express](url) and yes it works fine installs fine however on my project Prisma and Prisma client both are locked to 5.0.0 and that still doesnt work however locking them to 5.2.0 seems to work fine so I'll probably do that for now.
I wonder if it was version specific issue?

Edit: I forgot to say on prisma generate I m seeing a download indication for prisma engines which Ive never seen before ```Downloading Prisma engines for Node-API for windows 95%```

**Reply by DeveloperParth** [reply]
> (I am using Windows... so I have been "volunteered" to try this out if we have a reproduction)

I think it all started for me when I pruned the cache I had also updated pnpm to 8.7.4 and it was (I think) first time running ```pnpm i``` after that I doubt this could be the issue as ive tried multiple pnpm versions to solve this.

As Ive replied as of now the bug only seems to persist on prisma version 5.0.0. If I change version to 5.0.0 in [https://github.com/prisma/prisma-examples/tree/latest/typescript/rest-express](url) it throws the same error something to note here is that prisma client version can be 5.0.0 it doesn't affect anything
